Remove unneeded namespacing in header files

RonEld opened the pull request:
Remove the `mbedtls` namesapcing in the `#include` in header files
Resolves #857

hanno-arm commented on ChangeLog:
```diff
@@ -1,5 +1,11 @@
+= mbed TLS x.x.x branch released xxxx-xx-xx
+
+Bugfix
+   * Fix namespacing in header files. REmove the `mbedtls` namespacing in
```
Typo

RonEld commented:
@hanno-arm Thank you for your review
I fixed the typo. Please approve

andresag01 commented on include/mbedtls/net.h:
```diff
@@ -20,11 +20,11 @@
-#include "mbedtls/net_sockets.h"
+#include "net_sockets.h"
 #if defined(MBEDTLS_DEPRECATED_WARNING)
 #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h"
```
Should we change this one as well?

RonEld replied:
@andresag01 Why do you think it shouldn't change? Although `net.h` is superseded, it can still be included.

andresag01 replied:
This was a long time ago, so what I probably meant is that in the message for this warning there is a `mbedtls/net_sockets.h`. Do we want to make that `net_sockets.h` as well? Its a really minor thing and certainly not a stopper, more of a question. I will accept the PR.

RonEld replied:
I think in the message, it's more appropriate to keep it as is, as it is intended for human readers, and it's informative with the current wording

sbutcher-arm replied:
I personally think it's better to keep it as it is.

andresag01 commented:
@RonEld: Thanks for looking into this again, but could you please rebase instead of creating a merge commit. For example, the diff now shows as if you had introduced a new config file and in general its not ideal to have these merge commits in the history.

RonEld commented:
@andresag01 I have removed the merge commit and rebased.
Please review

RonEld commented:
backports created in #1779 and #1780 for mbedtls-2.1 and mbedtls-2.7 respectively

andresag01 commented on include/mbedtls/net.h:
```diff
@@ -1,7 +1,7 @@
  * \file net.h
  *
- * \brief Deprecated header file that includes mbedtls/net_sockets.h
+ * \brief Deprecated header file that includes net_sockets.h
```
Minor: Why did you remove `mbedtls/` from this comment for example and not from the one below?

RonEld replied:
I don't really have a good answer.
Probably because in this comment, it's more reasonable that a human reader will read the file within the same folder, and in the warning message, the compilation will probably be from the root folder. I'm fine either way

andresag01 commented on configs/config-ccm-psk-tls1_2.h:
```diff
@@ -83,6 +83,6 @@
 #define MBEDTLS_SSL_MAX_CONTENT_LEN             512
 
-#include "mbedtls/check_config.h"
+#include "check_config.h"
```
The README.txt file in the configs directory states that there are two options to use these config files
>1. Replace the default file include/mbedtls/config.h with the chosen one.
   (Depending on your compiler, you may need to adjust the line with
   #include "mbedtls/check_config.h" then.)
>
>2. Define MBEDTLS_CONFIG_FILE and adjust the include path accordingly.

So in case (1) the user has to do nothing and just copy the file over as stated in the readme (actually, I think the information in brackets is no longer relevant). However, in the case of (2) the user will probably need to add `./configs` to the include path as well as `./include/mbedtls` (because this patch removes the namespacing bit). Also, note that without this change, case (1) will also work with the `mbedtls/` prefix, its just not ideal and its already properly documented.

So I am not too sure that we should remove the namespacing from these config files. But if we do want to remove, we should at least fix the documentation accordingly.

RonEld replied:
I'll revert the changes in the configs

RonEld commented:
Added the "needs backports" label, as the backports are still pending reviews

RonEld commented:
@andresag01 I have reverted the changes in the `configs` folder, and added an `mbedtls` namespace in the `config-no-entropy.h` file as well. Please review

cc @hanno-arm

andresag01 commented on configs/config-no-entropy.h:
```diff
@@ -87,6 +87,6 @@
 #define MBEDTLS_AES_ROM_TABLES
 
-#include "check_config.h"
+#include "mbedtls/check_config.h"
```
I think this might not be ideal. The issue is that this file is copied by some import script directly into the inc/mbedtls directory in Mbed TLS, so technically the `mbedtls/` prefix is not needed.

RonEld commented:
@andresag01 I have removed the `mbedtls` namespacing as you requested.
I will update the backports once this PR is approved

sbutcher-arm commented:
@RonEld - Can you please backport the review feedback? #1779 and #1780 need the same fixes following the review of this PR.

RonEld commented:
Done

sbutcher-arm commented:
retest
